### DISCUSSION TITLE
Crabby Language

### DISCUSSION THREAD

**Discussion by TCROC** [crablang/crab#94] [Ideas]
A higher level easier to use language.  This one is ambitious and will probably consist of some R & D fun to start.  One of the most beautiful things about crab is its memory management and thread safety.  It would be amazing if a simpler to use strongly typed scripting language such as c# operated with with entirely coerced lifetimes (by default at least unless explicitly typed) and coerced borrows / moves / mutations (also by default unless explicitly typed) based on how the code is used.  Pretty much flipping all of the explicitness of rust on its head.  Would be a great gateway to bringing on other developers and teaching them the ways of the crab :)

---

Edit:

Currently the favored name is ``shrimp``!

```
crab -> shrimp
crabc -> shrimpc
crabgo -> shrimpgo
crabby -> shrimpy
crabfmt -> shimpfmt
crabup -> shrimpup
```

**Comment by TCROC** [comment]
We need a new name. Crabby is taken for clippy.

Hmm...

Spinoff to new sea creature?

Crayfish?

Then follow suit with:

crab -> cray
crabc -> crayc
crabgo -> craygo
crabby -> um... crayy? Oh I know: craycray
crabfmt -> crayfmt
crabup -> crayup

**Comment by TCROC** [comment]
Or lobster

crab -> lob
crabc -> lobc
crabgo -> lobgo
crabby -> lobby
crabfmt -> lobfmt
crabup -> lobup

**Comment by trvswgnr** [comment]
i think this is overall a great idea, but should definitely be its own repo 😉

**Comment by TCROC** [comment]
> i think this is overall a great idea, but should definitely be its own repo 😉

Yep I agree with this 100%

**Comment by JamalLyons** [comment]
I like this idea!

**Comment by KAVA-Leigh** [comment]
> Or lobster

Lobster already exists as a language :)

**Comment by TCROC** [comment]
> > Or lobster
> 
> Lobster already exists as a language :)

my day has been ruined... :(

is okay.  we just need a new sea creature... shrimp?

crab -> shrimp
crabc -> shrimpc
crabgo -> shrimpgo
crabby -> shrimpy
crabfmt -> shimpfmt
crabup -> shrimpup

**Comment by KAVA-Leigh** [comment]
Sorry!

Shrimp is good 👍🏻 

A baby crab is a Zoea, but that is far from catchy..

I was thinking along the lines of Squid, Starfish or Cod, but I think Shrimp is a clear winner here.

I like Baby Shark, but that may lead to legal issues lol

**Comment by TCROC** [comment]
> I like Baby Shark, but that may lead to legal issues lol

hahahaha that would've been great! 🤣

Ok cool I'll go edit the issue with "shrimp" being the favored name so far 🍤

**Comment by dabreyna** [comment]
Why not only shark? 🤣

**Comment by TCROC** [comment]
> Why not only shark?

I'm not apposed to this 🤔.  I kinda like it actually.  We have 2 good contenders now! :)

I believe @trvswgnr also suggested scampi in the Discord server

**Comment by Cuteistfox** [comment]
+1 for crab(as a difrent git branch named "more craby")
